numpy/distutils/lib2def.py

```diff
@@ -59,11 +59,11 @@ def parse_cmd():
         deffile = None
     return libfile, deffile
 
-def getnm(nm_cmd = ['nm', '-Cs', 'python%s.lib' % py_ver]):
+def getnm(nm_cmd=['nm', '-Cs', 'python%s.lib' % py_ver], shell=True):
     """Returns the output of nm_cmd via a pipe.
 
 nm_output = getnam(nm_cmd = 'nm -Cs py_lib')"""
-    f = subprocess.Popen(nm_cmd, shell=True, stdout=subprocess.PIPE, universal_newlines=True)
+    f = subprocess.Popen(nm_cmd, shell=shell, stdout=subprocess.PIPE, universal_newlines=True)
     nm_output = f.stdout.read()
     f.stdout.close()
     return nm_output
```

Comment:
These good lines should probably become `check_output`, so we don't ignore failures.

Reply:
Changed the default nm, could just remove the shell argument completely if we consider this private API though.

Reply:
I suppose `check_output` would result in different handling of `nm_err`.

Reply:
True, I suppose it could mean that someone has to use a try/except clause in theory. But it may also catch issues in practice? If we want to backport this, could be better to just leave it in place though?

Reply:
I do not know distutils well enough to make a judgement call about whether or not such a change is fine. It seems fine to me in the regard that it should be fairly straight forward to track down by someone skilled in basic python debugging... @rgommers thoughts?

Reply:
I don't think it matters. `check_output` would put it on stdout, this puts it in the exception message, which is perhaps nicer.

Using `subprocess.run` might be a little tidier while keeping the behavior, but it doesn't really matter.

Reply:
I changed this to capture and discard stderr (if returncode is OK) since running nm on the 32-bit `*lib` generates lots of warnings on stderr. In fact, the `nm_output` returned by this is not correctly parsed by the regex, so at least on my machine the `*.DEF` file and the subsequent `*.a` generated is not at all useful. But fixing that should be a seperate PR.

Reply:
Edit "running the 64-bit nm on the 32-bit `*.lib`"